### PR TITLE
Implement EinsumDense layer

### PR DESCRIPTION
**Pull request by pforderique**
Implements the `EinsumDense` layer from [Keras](https://github.com/keras-team/keras/blob/v2.13.1/keras/layers/core/einsum_dense.py#L32-L212) core. 

Depends on #7842 .

### CODE REVIEW

**Comment by mattsoulanille** [comment]
Before I review it, I just want to check if this PR is also intended to include CachedMultiHeadAttention and other layers besides EinsumDense. Thanks!

File: tfjs-layers/src/layers/nlp/einsum_dense_test.ts [hunk 1]
```diff
@@ -0,0 +1,317 @@
...
+        expect(layer.bias.shape).toEqual(combo.expectedBiasShape);
+      }
+      expect(outputTensor.shape).toEqual(combo.expectedOutputShape);
+    });
+  }
+
+  for (const combo of combinations) {
+    testWeightShape(combo);
+    testLayerCreation(combo);
+  }
```
Comment (mattsoulanille) [inline tfjs-layers/src/layers/nlp/einsum_dense_test.ts:298]: Nice!

**Review by mattsoulanille — APPROVED** [review]
LGTM

**Comment by mattsoulanille** [comment]
@Linchenn This one's ready for you to take a look. Thanks!

**Review by Linchenn — APPROVED** [review]

**Pull request merged**